views/default/elgg.js.php

```diff
@@ -65,8 +65,17 @@
 elgg.security.interval = <?php echo (int)_elgg_services()->actions->getActionTokenTimeout() * 333; ?>;
 elgg.config.language = '<?php echo (empty($CONFIG->language) ? 'en' : $CONFIG->language); ?>';
 
-// We require jquery-ui because loading the file in an AMD environment doesn't automatically
-// apply its functionality to jQuery.
+// jQuery and UI must be loaded sync in 2.x but modules should depend on these AMD modules
+define('jquery', function () {
```

Comment:
To be clear, what's driving all this is that some Elgg components (and probably 3rd parties) need jQuery UI loaded synchronously, before require can load them.

Currently we cheat a bit by SCRIPT loading them before elgg.js. jQuery core's module setup is flexible allowing both synchronous defining of the `jQuery` global var as well as auto-creating an AMD definition for "jquery".

jQuery UI's module setup is dumber: it just checks if `define` exists, and assumes if it does that you don't want the global `jQuery` var extended.